src/librustc_mir/interpret/operand.rs

```diff
@@ -132,6 +144,36 @@ impl<Tag> Operand<Tag> {
 
         }
     }
+
+    /// Returns a list of all `AllocId`s present in this `Operand`.
+    pub(super) fn alloc_ids(&self) -> SmallVec<[AllocId; 2]>
```

Comment:
Could you move this into `snapshot.rs`? I surely hope nothing else will ever need this kind of operation.^^